File: clients/php/tests/push.php
<?php
use RedisPriorityQueue as RQP;

// Requires
require_once(__DIR__.'/../utils/redis_instance.php');
require_once(__DIR__.'/../utils/const.php');
require_once(__DIR__.'/../lib/Lua.php');
require_once(__DIR__.'/../lib/Queue.php');

// Create Queue instance with Redis connection
$q = new RQP\Queue($r->getConnection());

// Set queue name
$q->setQueueName(QUEUE_NAME);

// Run
echo "* push (without priority):\n";
$res = $q->push(mt_rand());
echo $res."\n";

// Run
echo "* push (with priority):\n";
$res = $q->push(mt_rand(), PRIORITY);
echo $res."\n";

File: clients/python/push.py
#!/usr/bin/env python3

# Packages
import redis, random
from lib.RpqQueue import RpqQueue
from lib.RpqLua import RpqLua
from utils.redis_vars import *

def generateRandom():
    '''Helper: get a random integer'''
    return random.randint(1, 100000);

def getItem():
    '''Helper: generate item name'''
    return 'item_' + str(generateRandom());

# Redis instance
# !!! set Redis credentials into utils/redis_vars.py
r = redis.StrictRedis(host=rHost, port=rPort, db=rDbnum, password=rAuth)

# Load LUA Script
RpqLua = RpqLua(r, '../../src/redis-priority-queue.lua');
queue = RpqLua.register();

# RpqQueue instance
RpqQueue = RpqQueue(queue);

# Set queue name
RpqQueue.setqueueName('simple_queue')

# Push an item

print ('* push (without priority):')
item = getItem() # Item we will push to the queue
print ('Pushing item "%s" to the queue' % (item))
print (RpqQueue.push(item))

print ('* push (with priority):')
item = getItem() # Item we will push to the queue
print ('Pushing item "%s" to the queue' % (item))
print (RpqQueue.push(item, 1000))